MapboxCoreNavigation/RouteStep.swift

```diff
@@ -38,4 +39,46 @@ extension RouteStep {
     open var lastInstruction: SpokenInstruction? {
         return instructionsSpokenAlongStep?.last
     }
+    
+    /**
+     Returns true if the current route step contains a tunnel.
+     */
+    var containsTunnel: Bool {
+        guard let intersections = intersections else { return false }
+        for intersection in intersections {
+            if intersection.outletRoadClasses?.contains(.tunnel) == true {
+                return true
+            }
+        }
+        return false
+    }
+    
+    /**
+     Returns a tunnel slice for the current route step coordinates
+     */
+    var tunnelSlice: Polyline? {
+        guard let coordinates = coordinates, let tunnelIntersection = tunnelIntersection else { return nil }
+        return Polyline(coordinates).sliced(from: tunnelIntersection.entry.location, to: tunnelIntersection.exit.location)
+    }
+    
+    /**
+     Returns the distance of a tunnel slice for the current route step
+     */
+    var tunnelDistance: CLLocationDistance? {
+        guard let tunnelIntersection = tunnelIntersection else { return nil }
+        return tunnelIntersection.entry.location.distance(to: tunnelIntersection.exit.location)
```

Comment:
This distance calculation is an absolute point to point distance. Instead, we should be using a function which calculates distance along a line. I believe `Polyline` is a struct and contains a distance variable. It may be possible to remove this and just use `tunnelSlice.distance`.